pkg/controller/scaledobject/scaledobject_controller.go

```diff
@@ -148,13 +148,25 @@ func (r *ReconcileScaledObject) Reconcile(request reconcile.Request) (reconcile.
 	}
 
 	reqLogger.V(1).Info("Detecting ScaleType from ScaledObject")
-	if scaledObject.Spec.ScaleTargetRef == nil || scaledObject.Spec.ScaleTargetRef.DeploymentName == "" {
+	var errMsg string
+	if scaledObject.Spec.ScaleTargetRef != nil {
+		if scaledObject.Spec.JobTargetRef == nil {
+			reqLogger.Info("Detected ScaleType = Deployment")
+			return r.reconcileDeploymentType(reqLogger, scaledObject)
+		}
+		errMsg = "Both ScaledObject.Spec.ScaleTargetRef and ScaledObject.Spec.JobTargetRef cannot be set in the same time"
+	} else if scaledObject.Spec.JobTargetRef != nil {
 		reqLogger.Info("Detected ScaleType = Job")
 		return r.reconcileJobType(reqLogger, scaledObject)
 	} else {
-		reqLogger.Info("Detected ScaleType = Deployment")
-		return r.reconcileDeploymentType(reqLogger, scaledObject)
+		errMsg = "ScaledObject.Spec.ScaleTargetRef or ScaledObject.Spec.JobTargetRef is not set"
```

Comment:
Since both are not configured:
```suggestion
		errMsg = "ScaledObject.Spec.ScaleTargetRef and ScaledObject.Spec.JobTargetRef are not set"
```

Reply:
The intention of the err message should be to tell user to set one of the option, both of them cannot be set. I am open to another wording though :)

Reply:
Good point!